federatedscope/core/worker/client.py

```diff
@@ -254,17 +254,27 @@ def callback_funcs_for_evaluate(self, message: Message):
         self.state = message.state
         if message.content != None:
             self.trainer.update(message.content)
-        test_metrics = self.trainer.evaluate()
-        for key in test_metrics:
-            logging.info(
-                'Client #{:d}: (Evaluation (test set) at Round #{:d}) {:s} is {:.6f}'
-                .format(self.ID, self.state, key, test_metrics[key]))
-        val_metrics = self.trainer.validate()
-        for key in val_metrics:
-            logging.info(
-                'Client #{:d}: (Evaluation (val set) at Round #{:d}) {:s} is {:.6f}'
-                .format(self.ID, self.state, key, val_metrics[key]))
-        metrics = {**test_metrics, **val_metrics}
+        test_metrics, val_metrics, train_metrics = {}, {}, {}
+        if 'test' in self._cfg.eval.split:
+            test_metrics = self.trainer.evaluate()
+            for key in test_metrics:
+                logging.info(
+                    'Client #{:d}: (Evaluation (test set) at Round #{:d}) {:s} is {:.6f}'
+                    .format(self.ID, self.state, key, test_metrics[key]))
+        if 'val' in self._cfg.eval.split:
+            val_metrics = self.trainer.validate()
```

Comment:
@yxdyc @DavdGao  Is is possible to unify the interface, that is to say, using `trainer.evaluate(target_data_split_name='valid')` in place of this

Reply:
@joneswong , it is feasible to add a new option `valid` for the parameter `target_data_split_name`, and I will push it in a new pr

Reply:
We have reserved such interfaces (see [evaluate](https://github.com/alibaba/FederatedScope/blob/master/federatedscope/core/trainers/trainer.py#L170) and [valid](https://github.com/alibaba/FederatedScope/blob/master/federatedscope/core/trainers/trainer.py#L185)).

Reply:
> 

So can we replace the function `validate` with `evaluate(target_data_split_name='val')`?

Reply:
> > 
> 
> So can we replace the function `validate` with `evaluate(target_data_split_name='val')`?

After further discussion, we agree that there is no need to reserve both `evaluate()` and `validate()` interfaces, since the behaviors when doing validation should be the same as the one for evaluation. 
Please @rayrayraykk unify them in this PR by the way.